src/test/java/mops/LayeredArchitectureTest.java

```diff
@@ -0,0 +1,49 @@
+package mops;
+
+import com.tngtech.archunit.junit.AnalyzeClasses;
+import com.tngtech.archunit.junit.ArchTag;
+import com.tngtech.archunit.junit.ArchTest;
+import com.tngtech.archunit.lang.ArchRule;
+import mops.businesslogic.DirectoryService;
+import mops.businesslogic.FileService;
+import mops.businesslogic.GroupService;
+import mops.persistence.FileRepository;
+
+import static com.tngtech.archunit.library.Architectures.layeredArchitecture;
+
+@ArchTag("checkLayeredArchitecture")
+@AnalyzeClasses(packages = "mops")
+public class LayeredArchitectureTest {
+
+    private static final String MOPS_PRESENTATION = "mops.presentation..";
+    private static final String MOPS_BUSINESSLOGIC = "mops.businesslogic..";
+    private static final String MOPS_PERSISTENCE = "mops.persistence..";
+    /**
+     * This checks, if the layer is correctly used and
+     * no wrong accesses are made.
+     */
+    @ArchTest
+    static final ArchRule checkLayeredArchitecture = layeredArchitecture()
+            .layer("mopsPersistence").definedBy(MOPS_PERSISTENCE)
+            .layer("mopsBusinesslogic").definedBy(MOPS_BUSINESSLOGIC)
+            .layer("mopsPresentation").definedBy(MOPS_PRESENTATION)
+
+            .whereLayer("mopsPresentation").mayNotBeAccessedByAnyLayer()
+            .whereLayer("mopsBusinesslogic").mayOnlyBeAccessedByLayers("mopsPresentation")
+            .whereLayer("mopsPersistence").mayOnlyBeAccessedByLayers(
+                    "mopsPresentation", "mopsBusinesslogic")
+
+            /*
+             * These dependencies are between the Tests and the different layers,
+             * which of course is correct and wanted and should not lead to a failing test.
+             */
+            .ignoreDependency(Material1ApplicationTests.class, DirectoryService.class)
+            .ignoreDependency(Material1ApplicationTests.class, FileRepository.class)
+            .ignoreDependency(Material1ApplicationTests.class, FileService.class)
+            .ignoreDependency(Material1ApplicationTests.class, GroupService.class)
+            .ignoreDependency(SecurityTests.class, DirectoryService.class)
+            .ignoreDependency(SecurityTests.class, FileRepository.class)
+            .ignoreDependency(SecurityTests.class, FileService.class)
+            .ignoreDependency(SecurityTests.class, GroupService.class)
```

Comment:
can we make this easier with some kind of `@LayerArchExclusion` annotation. (name still open for suggestions)

Reply:
Technically, yes, but I thought that this would be way clearer and it's not like it's creating unreadable code or messy code or anything of that sort.

Reply:
this is a prime example of error prone and hard to read code

Reply:
can you try to use `importOptions = ImportOption.DoNotIncludeTests.class` in the `@AnalyzeClasses` annotation instead if that is easier?

Reply:
Sadly, this does not work and just causes errors. This is the best and easiest solution, even though it might not look as good as your recommendation...

Reply:
In the end it did work, turns out I wasn't using it correctly.